src/metabase/api/caching.clj

```diff
@@ -0,0 +1,150 @@
+(ns metabase.api.caching
+  (:require
+   [clojure.walk :as walk]
+   [compojure.core :refer [GET]]
+   [metabase.api.common :as api]
+   [metabase.api.common.validation :as validation]
+   [metabase.db.query :as mdb.query]
+   [metabase.events :as events]
+   [metabase.public-settings.premium-features :as premium-features :refer [defenterprise]]
+   [metabase.util :as u]
+   [metabase.util.i18n :refer [tru]]
+   [metabase.util.malli.schema :as ms]
+   [toucan2.core :as t2]))
+
+;; Data shape
+
+(defn row->config
+  "Transform from how cache config is stored to how it's used/exposed in the API"
+  [row]
+  {:model    (:model row)
+   :model_id (:model_id row)
+   :strategy (assoc (:config row) :type (:strategy row))})
+
+(defn config->row
+  "Transform cache config from API form into db storage from"
+  [{:keys [model model_id strategy]}]
+  {:model    model
+   :model_id model_id
+   :strategy (:type strategy)
+   :config   (dissoc strategy :type)})
+
+(defn- drop-internal-fields
+  "See `metabase-enterprise.caching.strategies/CacheStrategy`"
+  [schema]
+  (walk/prewalk
+   (fn [x]
+     (if (and (vector? x) (= (first x) :map))
+       (into [] (remove #(:internal (meta %))) x)
+       x))
+   schema))
+
+;; TODO: declaring schema as defenterprise will make OpenAPI docs invalid for an EE version
+(defenterprise CacheStrategy
```

Comment:
I'm confused by this comment.. have we decided it's OK to break the OpenAPI docs for EE for now?

Reply:
do you like the new comment better? :)

Reply:
much clearer, thanks!